packages/dashboard/src/customization/widgets/lineSymbol/component.test.tsx

```diff
@@ -0,0 +1,221 @@
+          end: { x: 180, y: 180 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dashed',
+          color: 'blue',
+          thickness: 10,
+          start: { x: 180, y: 175 },
+          end: { x: 20, y: 30 },
+        } as LineProperties,
+        isSelected: true,
+      },
+      {
+        properties: {
+          lineStyle: 'dotted',
+          color: 'green',
+          thickness: 15,
+          start: { x: 180, y: 175 },
+          end: { x: 20, y: 30 },
+        } as LineProperties,
+        isSelected: true,
+      },
+    ].forEach((configuration) => {
+      const widget = { ...mockWidgetPartial, properties: { ...configuration.properties } };
+      const isSelected = configuration.isSelected;
+      it(`should render with ${JSON.stringify(widget)} correctly. The widget ${
+        isSelected ? 'is selected' : 'is not selected'
+      }.`, () => {
+        const { container } = render(<TestComponent widget={widget} isSelected={isSelected} />);
+        expect(container).toMatchSnapshot();
+      });
+    });
+  });
+  // make test component wrapped in all the providers
+  describe('Anchor visibility', () => {
+    const widget = {
+      ...mockWidgetPartial,
+      properties: {
+        lineStyle: 'solid',
+        color: 'black',
+        thickness: 5,
+        start: { x: 180, y: 175 },
+        end: { x: 20, y: 30 },
+      },
+    } as LineWidget;
+
+    it('should display anchors when the widget is selected', async () => {
+      // (useIsSelected as jest.Mock).mockReturnValue(true); // this makes it work, but we shouldn't need it
+      const { getByTestId, findByTestId } = render(<TestComponent widget={widget} isSelected={true} />);
+      expect(getByTestId('line-start-anchor')).toBeInTheDocument();
+      expect(getByTestId('line-end-anchor')).toBeInTheDocument();
+    });
+    it('should not display anchors when the widget is not selected', () => {
+      // (useIsSelected as jest.Mock).mockReturnValue(false);
+      const { queryByTestId } = render(<TestComponent widget={widget} isSelected={false} />);
+      expect(queryByTestId('line-start-anchor')).not.toBeInTheDocument();
+      expect(queryByTestId('line-end-anchor')).not.toBeInTheDocument();
+    });
+  });
+  describe('manipulating the widget', () => {
+    const beforeChangeWidget = {
+      ...mockWidgetPartial,
+      properties: {
+        lineStyle: 'solid',
+        color: 'black',
+        thickness: 5,
+        start: { x: 50, y: 50 },
+        end: { x: 100, y: 100 },
+      },
+    } as LineWidget;
+
+    const renderWithStore = (initialState: Partial<DashboardState>) => {
+      const store = configureDashboardStore(initialState);
+
+      const TestComponentWithStore = () => (
+        <Provider store={store}>
+          <DndProvider backend={HTML5Backend}>
+            <LineWidgetComponent {...(store.getState().dashboardConfiguration.widgets[0] as LineWidget)} />
```

Comment:
Be careful on this line, the widget will never be updated when the store is updated. you would need to use the redux selector for that to happen

Reply:
done